File: csharp/booking-up-for-beauty/BookingUpForBeauty.cs
static class Appointment
{
    public static DateTime Schedule(string appointmentDateDescription) => DateTime.Parse(appointmentDateDescription);
    public static bool HasPassed(DateTime appointmentDate) => DateTime.Now > appointmentDate;
    public static bool IsAfternoonAppointment(DateTime appointmentDate) => appointmentDate.Hour >= 12 && appointmentDate.Hour < 18;

    //Have I become obsessed with method-bodied members?
    public static string Description(DateTime appointmentDate) =>
        "You have an appointment on " + appointmentDate.ToString("M/d/yyyy h:mm:ss tt") + ".";

    //Le Gasp! :o
    //The Readme has been giving answers in method-bodied member notation! (kinda)
    public static DateTime AnniversaryDate() => new DateTime(DateTime.Now.Year, 9, 15, 0, 0, 0);
}

File: go/booking-up-for-beauty/booking_up_for_beauty.go
package booking

import (
	"fmt"
	"time"
)

// Schedule returns a time.Time from a string containing a date
func Schedule(date string) time.Time {
	d, _ := time.Parse("1/2/2006 15:04:05", date)
	return d
}

// HasPassed returns whether a date has passed
func HasPassed(date string) bool {
	t, _ := time.Parse("January 2, 2006 15:04:05", date)
	return t.Before(time.Now())
}

// IsAfternoonAppointment returns whether a time is in the afternoon
func IsAfternoonAppointment(date string) bool {
	t, _ := time.Parse("Monday, January 2, 2006 15:04:05", date)
	return t.Hour() >= 12 && t.Hour() < 18
}

// Description returns a formatted string of the appointment time
func Description(date string) string {
	t, _ := time.Parse("1/2/2006 15:04:05", date)
	//You have an appointment on Monday, June 6, 2005, at 10:30.
	return fmt.Sprintf("You have an appointment on %v, %v %d, %d, at %d:%d.", t.Weekday(), t.Month(), t.Day(), t.Year(), t.Hour(), t.Minute())
}

// AnniversaryDate returns a Time with this year's anniversary
func AnniversaryDate() time.Time {
	return time.Date(time.Now().Year(), time.September, 15, 0, 0, 0, 0, time.UTC)
}
